Tools/CMake/AMReXInstallHelpers.cmake

```diff
@@ -23,25 +23,29 @@ function ( generate_amrex_config_header )
 
    endforeach ()
 
-   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" )
```

Comment:
If you want to avoid code nesting in vars and nesting of `if`s it could be so easy:
```cmake
       if (ALLOW_DIFFERENT_COMPILER)
           set(COMPILER_ID_MACRO __cplusplus)
       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" )
           set(COMPILER_ID_MACRO __GNUC__)
```